### ISSUE TITLE
Add CMake support

### ISSUE THREAD

**Issue by egorodet** [wolfpld/tracy#2]
Since Tracy is a cross-platform project that targets multiple operating and build systems it is highly required to support CMake build and integration with other CMake-based projects. While it's generally easy to add `TracyClient.cpp` to the project build, it may look dirty in CMakeList.txt to add some external file to your module and it would be much cleaner to add static linking with externally defined `TracyClient` CMake target.

**Comment by neobrain** [comment]
I've written an (unfinished) CMakeLists.txt for packaging Tracy via Conan: https://gist.github.com/neobrain/40b010846b32ddd463de273e12cd9663

I agree having out-of-the-box CMake support would be wonderful. If the author is interested in adding this, I'm happy to clean up the file linked above and create a PR for it.

**Comment by wolfpld** [comment]
If you are willing to document it in the manual and keep support it later, then sure, go ahead.

Just don't change the existing build system.

**Comment by cliffowen** [comment]
I'd like to +1 this request. I added my own cmake files but it would be nice if they were the default solution. FWIW: I also build to a static library rather than including a header. Rather than this being a clean/dirty approach, the reality is that it's an external file that rarely changes but it affects my build times to always build it and my build times are as critical for the project as the execution speed of the project. If it adds even 1 second to recompile the same code over and over, it's moved into a lib.

**Comment by Manu343726** [comment]
@neobrain I've also added cmake for conan support in my fork: https://github.com/Manu343726/tracy

Is your package available somewhere public? Mine is here: https://bintray.com/manu343726/conan-packages. Maybe we could talk to agree on a Cmake support PR for tracy and add an official conan recipe to [conan-center-index](https://github.com/conan-io/conan-center-index) (@wolfpld you don't have to worry about conan support, recipes should be non-intrusive and are hosted and maintained in the official index repo).

**Comment by tonka3000** [comment]
Would love to see tracy in [conan-center-index](https://github.com/conan-io/conan-center-index) as well. It's an awesome project.

**Comment by Manu343726** [comment]
@tonka3000 https://github.com/conan-io/conan-center-index/issues/1708

@wolfpld could you please review my CMake config? https://github.com/Manu343726/tracy/blob/master/CMakeLists.txt

If you think it is ok I will PR it to your master branch. (CMake port of the profiler will come later)

**Comment by wolfpld** [comment]
You don't need to explicitly specify `TRACY_PORT`.

**Comment by wolfpld** [comment]
And maybe it would be better to allow control of both data and broadcast port? You usually don't want to change the broadcast port.

```
If for some reason you want to use another port , you can change it using the TRACY_DATA_PORT macro
for the data connection, and TRACY_BROADCAST_PORT macro for client broadcasts. Alternatively, both ports
may be changed at the same time by declaring the TRACY_PORT macro (specific macros listed before have
higher priority). The data connection port may be also changed without recompiling the client application,
by setting the TRACY_PORT environment variable.
```

**Comment by wolfpld** [comment]
Fixed by #247.
